src/main/java/uk/gov/hmcts/reform/userprofileapi/service/impl/IdamServiceImpl.java

```diff
@@ -107,22 +134,25 @@ public AttributeResponse updateUserDetails(UpdateUserDetails updateUserDetails,
                     null));
         } catch (FeignException ex) {
             log.error("{}:: {} {}", loggingComponentName, "SIDAM call failed:", ex);
-            responseEntity = ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).build();
+            responseEntity = ResponseEntity.status(HttpStatus.UNAUTHORIZED).build();
         }
         return new AttributeResponse(responseEntity);
     }
 
     public HttpStatus gethttpStatusFromFeignException(FeignException ex) {
         return (ex instanceof RetryableException)
-                ? HttpStatus.INTERNAL_SERVER_ERROR
+                ? HttpStatus.UNAUTHORIZED
                 : HttpStatus.valueOf(ex.status());
     }
 
     @SuppressWarnings("unchecked")
     private IdamRolesInfo buildIdamResponseResult(Response response) {
+
         ResponseEntity<Object> entity = JsonFeignResponseHelper.toResponseEntity(response,
                 getResponseMapperClass(response, IdamUserResponse.class));
-        return new IdamRolesInfo(entity);
+        IdamRolesInfo idamRolesInfo = new IdamRolesInfo(entity);
+
+        return idamRolesInfo;
```

Comment:
Any reason for this  change.Can't this be the existing code
`return new IdamRolesInfo(entity);`